Gems pfizer pov (#659)
SimpleDataLabsInc/prophecy-docs: docs/analysts/development/gems/source-target/external-table/oracle.md

---
title: Oracle
id: oracle
slug: /analysts/oracle
description: Read and write from Oracle
tags: []
---

<span class="badge">Prophecy Automate</span><br/><br/>

Use a table from Oracle as an external source.

:::note
Oracle target not supported.
:::

## Source parameters

| Parameter                   | Tab             | Description                                                       |
| --------------------------- | --------------- | ----------------------------------------------------------------- |
| Connection type             | Type            | Location you want to connect from.                                |
| Format type                 | Source location | Format of the gem. In this case, `oracle`.                        |
| Select or create connection | Source location | Whether to select an existing connection, or to create a new one. |
| Read using                  | Source location | Whether to read with tables or queries.                           |

## Read using table

> When you read with a table, you need to provide the:
> 
> - **Schema**: Schema where the table is located.
> - **Name**: Name of the external table to read.

Reviewer: ```suggestion
When you read with a table, you need to provide the following properties:

| Property | Description |
| ---------| ------------ |
| Schema | Schema where the table is located. |
| Name | Name of the external table to read. |
```

Suggestion: List the properties in a table since all the other gem pages have it that way.

## Read using query

> When you read using a query, you can write a query directly in the source gem to retrieve the correct table.

Reviewer: ```suggestion
If you want to read with a query, provide a query in the source gem to retrieve the correct table.
```

Suggestion: Saying "write" is a bit confusing here because I thought it meant to write to a table. But I rewrote the sentence to make it clearer based on what it originally sounded like -- but it also leaves me wondering WHERE do I put the query?

Author: should be clear in the UI. trying to avoid a screenshot. i suppose i can say in the "query" field.
